src/editor/operations.ts

```diff
@@ -205,8 +209,18 @@ export function formatRangeAsCode(range: Range): void {
             parts.push(partCreator.newline());
         }
     } else {
-        toggleInlineFormat(range, "`");
-        return;
+        const fenceLen = longestBacktickSequence(range.text);
+        if (hasBacktick) {
+            parts.unshift(partCreator.plain("`".repeat(fenceLen + 1)));
+            parts.push(partCreator.plain("`".repeat(fenceLen+ 1)));
+        } else {
+            if (fenceLen === 0) {
+                toggleInlineFormat(range, "`");
+                return;
+            }
+            toggleInlineFormat(range, "`".repeat(fenceLen));
+            return;
+        }
```

Comment:
This works but I think can be simplified, can you not call `toggleInlineFormat` in all branches here?

Reply:
> This works but I think can be simplified, can you not call `toggleInlineFormat` in all branches here?

Unfortunately, that's the only way i could come up with, as stated earlier, i tried to simplify further but it doesn't work which prompted me to leave it this way

Reply:
I simplified it locally to 

![image](https://user-images.githubusercontent.com/2403652/162731035-14d4e188-71f4-4862-b110-a567a5654a08.png)

Maybe take some inspiration from this?

Reply:
Alright thanks. I'll do that ASAP. Also suddenly some tests are failing now when i push here but it all works fine on my local machine when i run `yarn test`.